src/AutoRest.CSharp/Common/Output/Models/Types/ModelFactoryTypeProvider.cs

```diff
@@ -88,6 +115,21 @@ private static string GetDefaultNamespace()
 
         public HashSet<MethodInfo> ExistingModelFactoryMethods { get; }
 
+        private SignatureType? _signatureType;
+        public override SignatureType? SignatureType
+        {
+            get
+            {
+                // This can only be used for Mgmt now, because there are custom/hand-written code in HLC can't be loaded into CsharpType such as generic methods
+                // TODO: enable this for DPG, and check Configuration.Generate1ConvenientClient to disable it for HLC
+                if (!Configuration.AzureArm)
+                {
+                    return null;
+                }
+                return _signatureType ??= new SignatureType(_typeFactory, ShouldNotBeUsedForOutput().Select(x => (MethodSignature)x.Signature).ToList(), _sourceInputModel, DefaultNamespace, DefaultName);
```

Comment:
we should not really use `DefaultNamespace` and `DefaultName` here.
DefaultNamespace and DefaultName are kind of like "requested" namespace and name.
The generator might not generate those namespaces or names, for instance, when there is an existing type with `CodeGenType`, and it has a different namespace and name. When this happens, the generator will just write the namespace and name of the existing type ignoring the default namespace and default name.

Here I believe we should use `Declaration.Namespace` and `Declaration.Name`.

Reply:
got it, thanks, updated.